src/awkward/_connect/avro.py

```diff
@@ -292,6 +292,7 @@ def rec_exp_json_code(
                     parameters={"__array__": "char"},
                     form_key=f"node{form_next_id+1}",
                 ),
+                parameters={"__array__": "string"},
```

Comment:
Avro deserialiser was missing `"string"` names

Reply:
I'm not surprised—this isn't an unlikely mistake. (We didn't spend much time on this module because it was a stepping-stone to the real problem.)